### PR TITLE
feat(cta-section): created component

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
### Related Ticket(s)

#5218 

### Description

Created `DDSCtaBlock` and changed `CTASection` to match the new specs.

image

### CODE REVIEW

**Comment by jeffchew** [comment]
@RaphaelAmadeu there are some conflicts, can you see if you can resolve?

**Comment by DragosRistici** [comment]
## A few small knob issues:


Web Components (https://ibmdotcom-webcomponents.s3.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/5767/index.html?path=/story/components-cta-section--default)

1. Icon change doesn't seem to work

Screenshot 2021-04-13 at 15 19 27


2. Copy text doesn't seem to change

Screenshot 2021-04-13 at 15 22 44

**Comment by oliviaflory** [comment]
Seeing some unexpected spacing that I want to double check with other designers before adding any comments.

1. Pictograms not visible on Gray 90 or Gray 100 themes
Screen Shot 2021-04-13 at 9 16 21 AM


2. Not seeing a focus state around the video players when tabbing with the keyboard
CTA-section-Apr-13-2021 09-19-05

**Comment by oliviaflory** [comment]
### CTA block
1. Heading should span 8 columns with 32px inset padding. Currently only spanning 5 columns?
Screen Shot 2021-04-13 at 9 41 20 AM


2. Copy should span 8 columns with 32px inset padding. Currently only spanning 5 columns?
Screen Shot 2021-04-13 at 9 41 34 AM

Expected CTA block to look like the current CTA section in Web components canary:
Screen Shot 2021-04-13 at 9 54 45 AM
Screen Shot 2021-04-13 at 9 54 59 AM




#### with content items
1. Three content items are squeezed into 8 columns and not aligned to the grid properly. They should span 4 columns each, 3 across 12 columns, or 2 across 8 columns and 1 falling onto a second line
Screen Shot 2021-04-13 at 9 42 27 AM


2. Storybook should be sentence case "With content items", not title case "With Content Items"

#### with link list 
2. Storybook should be sentence case "With link list", not title case "With Link List"

File: packages/web-components/src/components/cta-section/cta-section.scss [hunk 1]
```diff
@@ -5,84 +5,30 @@
...
 
-  @include carbon--breakpoint('sm') {
-    padding-top: $carbon--layout-03;
-  }
-
-  @include carbon--breakpoint('lg') {
-    padding-top: $carbon--layout-05;
+  .#{$prefix}--content-section {
+    margin-left: -1rem;
+    margin-right: -1rem;
```
Comment (kennylam) [inline packages/web-components/src/components/cta-section/cta-section.scss:16]:
```suggestion
    @include hang;
```

**Review by kennylam — COMMENTED** [review]
Looks good @RaphaelAmadeu! Just a few small things.

File: packages/web-components/examples/codesandbox/components/cta-section/index.html [hunk 2]
```diff
@@ -24,17 +24,139 @@
...
+      <dds-cta-block-item>
+        <svg
+          slot="media"
+          focusable="false"
+          preserveAspectRatio="xMidYMid meet"
+          xmlns="http://www.w3.org/2000/svg"
+          aria-label="Pictogram description"
+          viewBox="0 0 32 32"
+          role="img"
+        >
```
[the remark this answers is not in the document]
Comment (kennylam) [reply packages/web-components/examples/codesandbox/components/cta-section/index.html:70]: @RaphaelAmadeu My mistake, I see this is the Codesandbox HTML. Disregard this comment, sorry!

**Comment by kennylam** [comment]
@RaphaelAmadeu you need to include `utils` in order to use `hang`, sorry if i wasn't clear on that.
```scss
@import '../../globals/utils/hang';
```

**Comment by kennylam** [comment]
> 2\. Not seeing a focus state around the video players when tabbing with the keyboard

@oliviaflory This issue is being tracked in https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5514.

File: packages/web-components/src/components/cta-section/__stories__/README.stories.mdx [hunk 3]
```diff
@@ -17,32 +16,158 @@ Here's a quick example to get you started.
...
+        <span slot="statistics">12%</span>
+        <dds-content-item-heading>Learn how</dds-content-item-heading>
+        <dds-content-item-copy>Dig into more self-directed larning about DevOps methodologies.</dds-content-item-copy>
+        <dds-text-cta slot="footer" cta-type="local" icon-placement="right" href="example.com">Browse tutorials</dds-text-cta>
+        <dds-text-cta slot="footer" cta-type="local" icon-placement="right" href="example.com">Browse tutorials</dds-text-cta>
+      </dds-cta-block-item>
+    </dds-cta-block-item-row>
+    <dds-cta-block-item-row>
+      <dds-cta-block-item>
+        <svg
```
Comment (jeffchew) [inline packages/web-components/src/components/cta-section/__stories__/README.stories.mdx:74]: @RaphaelAmadeu it looks like the API for CTA Section has changed, can you add in as a breaking change under the pinned issue? We'll need to probably call this out in our announcement too...

File: packages/web-components/src/components/promo-group/__stories__/promo-group.stories.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /**
  * @license
  *
- * Copyright IBM Corp. 2020
+ * Copyright IBM Corp. 2020, 2021
```
Comment (jeffchew) [inline packages/web-components/src/components/promo-group/__stories__/promo-group.stories.ts:4]: @RaphaelAmadeu is promo-group / promo-item supposed to be deleted?

Comment (andysherman2121) [reply packages/web-components/src/components/promo-group/__stories__/promo-group.stories.ts:4]: I thought it was supposed to be deleted

File: packages/web-components/src/components/cta-section/__stories__/README.stories.mdx (hunk 3 above)
Comment (jeffchew) [reply packages/web-components/src/components/cta-section/__stories__/README.stories.mdx:74]: I'm wondering even if a note should be added to the README that the API for CTA Section has changed as of v1.18.0, thoughts @RaphaelAmadeu ?

**Comment by jeffchew** [comment]
@RaphaelAmadeu there are a number of stories in dotcom shell, etc that are using the old API for CTA Section that needs updating:

image

**Comment by shixiedesign** [comment]
I will reserve the storybook issues for later post freeze, here are the component styling issues I noticed: 

1. Spacing between pictogram and the content under should be 32px 
Current:
image
 Expected:
image

2. Same height util didn't work. Expecting the link CTAs to be top aligned
image

File: packages/web-components/src/components/cta-section/__stories__/README.stories.mdx (hunk 3 above)
Comment (jeffchew) [reply packages/web-components/src/components/cta-section/__stories__/README.stories.mdx:74]: Perhaps add a section like this in the README:

```
## Breaking Changes
  
As of `v1.18.0`, the API for `<dds-cta-section>` has changed. Be sure to update to the latest version of CTA Section if upgrading from a previous version.
```

**Comment by jeffchew** [comment]
@RaphaelAmadeu CI checks are failing for web components:

image

**Review by shixiedesign — APPROVED** [review]
YAAAAS!

**Review by jeffchew — APPROVED** [review]
Thank you @RaphaelAmadeu!!

**Pull request merged**